### PR TITLE
V3 0 0 component

### PR DESCRIPTION
**Pull request by jcscottiii**
This depends on #159 
This patch introduces the following fields to the component.yaml: `parameters` `responsible_role`
This patch modifies the following fields: `narrative`

`narrative` now contains two fields: `key` and `text`. the `key` field is optional
`parameters` now contains two fields: `key` and `text`. both fields are required if used.
`responsible_role` is a single string.

These new fields are what make the 3.0.0 version schema of component.yaml

This PR does not change currently how the information is rendered to gitbook or docx. That will come in a subsequent PR.

An example schema for 3.0.0 is as follows:
```yaml
documentation_complete: false
name: Amazon Elastic Compute Cloud
key: EC2
references:
- name: Reference
  path: http://VerificationURL.com
  type: URL
satisfies:
- control_key: CM-2
  covered_by:
  - verification_key: EC2_Verification_1
  - component_key: UAA
    system_key: CloudFoundry
    verification_key: UAA_Verification_1
  implementation_status: partial
  narrative:
    - key: "a"
      text: "Justification in narrative form A for CM-2"
    - key: "b"
      text: "Justification in narrative form B for CM-2"
  standard_key: NIST-800-53
- control_key: 1.1
  covered_by:
  - verification_key: EC2_Verification_1
  - component_key: UAA
    system_key: CloudFoundry
    verification_key: UAA_Verification_1
  implementation_status: partial
  parameters:
    - key: "a"
      text: "Parameter A for 1.1"
    - key: "b"
      text: "Parameter B for 1.1"
  narrative:
    - key: "a"
      text: "Justification in narrative form A for 1.1"
    - key: "b"
      text: "Justification in narrative form B for 1.1"
  standard_key: PCI-DSS-MAY-2015
- control_key: 2.1
  covered_by: []
  implementation_status: partial
  narrative:
    - text: "Justification in narrative form for 2.1"
  standard_key: PCI-DSS-MAY-2015
responsible_role: "AWS Staff"
schema_version: 3.0.0
verifications:
- key: EC2_Verification_2
  name: EC2 Governor 2
  path: artifact-ec2-1.png
  type: Image
- key: EC2_Verification_1
  name: EC2 Verification 1
  path: http://VerificationURL.com
  type: URL

```

### CODE REVIEW

**Comment by afeld** [comment]
Is there a corresponding issue(s) for this?

File: docx/docx.go [hunk 1]
```diff
@@ -45,7 +45,9 @@ func (openControl *OpenControlDocx) FormatControl(standardControl string) string
 		for _, justification := range selectJustifications {
 			openControl.Components.GetAndApply(justification.ComponentKey, func(component base.Component) {
 				text = fmt.Sprintf("%s%s  \n", text, component.GetName())
-				text = fmt.Sprintf("%s%s  \n", text, justification.SatisfiesData.GetNarrative())
+				for _, narrative := range justification.SatisfiesData.GetNarratives() {
+					text = fmt.Sprintf("%s%s  \n", text, narrative.GetText())
```
Comment (afeld) [inline docx/docx.go]: Minor: any need to keep the trailing whitespace?

File: models/components/versions/3_0_0/component_test.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+func TestComponentGetters(t *testing.T) {
+	testSatisfies := []Satisfies{{Parameters: []Section{Section{Key:"key", Text: "text"}}, Narrative: []NarrativeSection{NarrativeSection{Key: "key", Text: "text"}, NarrativeSection{Text: "text"}}}, {}, {}, {}}
+	component := Component{
+		Name:          "Amazon Elastic Compute Cloud",
+		Key:           "EC2",
+		ResponsibleRole: "AWS Staff",
+		References:    common.GeneralReferences{{}},
+		Verifications: common.VerificationReferences{{}, {}},
+		Satisfies:     testSatisfies,
+		SchemaVersion: semver.MustParse("2.0.0"),
```
Comment (afeld) [inline models/components/versions/3_0_0/component_test.go]: Should this be `3.0.0`?

**Comment by afeld** [comment]
Looks great! Just the minor things.

Have you tried running this over any of the real-world V3 formatted YAML files, e.g. https://github.com/18F/cg-compliance/pull/86?

File: docx/docx.go (hunk 1 above)
Comment (jcscottiii) [inline docx/docx.go]: For now, yeah. I don't want to change any rendering output in this PR. All those rendering changes will be in #161.

File: models/components/versions/3_0_0/component_test.go (hunk 2 above)
Comment (jcscottiii) [inline models/components/versions/3_0_0/component_test.go]: lol probably. doh.

**Comment by jcscottiii** [comment]
@afeld going to try it with #161 when i finish that one

**Comment by jcscottiii** [comment]
@afeld tried it, there's some issues with the component.yaml. But right now, you can't see it. (they just don't get surfaced to the console). created https://github.com/opencontrol/compliance-masonry/issues/163. PR to come after #161

**Comment by afeld** [comment]
Looks great! Do we want to merge to `master` now, or wait for manual testing w/ #161?

**Comment by jcscottiii** [comment]
@afeld we can merge this one. lets hold on #161 though

**Pull request merged**